compaction/compaction_manager.cc

```diff
@@ -755,7 +760,7 @@ future<> compaction_manager::rewrite_sstables(column_family* cf, sstables::compa
 }
 
 future<> compaction_manager::perform_sstable_scrub_validate_mode(column_family* cf) {
-    return run_custom_job(cf, sstables::compaction_type::Scrub, [this, &cf = *cf, sstables = get_candidates(*cf)] () mutable -> future<> {
+    return run_custom_job(cf, sstables::compaction_type::Scrub, run_exclusively::yes, [this, &cf = *cf, sstables = get_candidates(*cf)] () mutable -> future<> {
```

Comment:
don't think we need to run validation with compaction stopped or exclusive lock, as it's a read-only operation. we can only retrieve all sstables in the set as done by my patch "compaction: don't rule out compacting sstables in validate-mode scrub"

Reply:
True, it's a read only operation, although it is  less useful if it detects validation errors in sstables that are later, or in parallel, compacted and go away.  To prevent that, the user can disable auto-compactiom before running scrub/validate.

Maybe as a follow up we need to quarantine sstables that failed validation, e.g by moving them into a `quarantine` sub-directory that holds sstables that are readable but protected from compaction.

Reply:
Agree with both your points. At this point, that should be our suggestion (i.e. disable auto compaction first) for everyone needing either scrub or validate.

Reply:
Given that we intend to move invalid sstables to quarantine, we can already avoid the lock path for validation. So it's less change in the future.

Reply:
Done in v2.